rust/cubesql/cubesql/src/compile/protocol.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{compile::CubeContext, CubeError};
+use datafusion::datasource;
+use std::{
+    fmt::Debug,
+    hash::{Hash, Hasher},
+    sync::Arc,
+};
+
+pub trait DatabaseProtocolDetails: Debug + Send + Sync {
+    fn get_name(&self) -> &'static str;
+
+    fn support_set_variable(&self) -> bool;
+
+    fn support_transactions(&self) -> bool;
+
+    fn get_provider(
+        &self,
+        context: &CubeContext,
+        tr: datafusion::catalog::TableReference,
+    ) -> Option<std::sync::Arc<dyn datasource::TableProvider>>;
+
+    fn table_name_by_table_provider(
+        &self,
+        table_provider: Arc<dyn datasource::TableProvider>,
+    ) -> Result<String, CubeError>;
+}
+
+impl PartialEq for dyn DatabaseProtocolDetails {
+    fn eq(&self, other: &Self) -> bool {
+        self.get_name() == other.get_name()
```

Comment:
Given that we are using `'static` trait objects, why not just rely on `Any::type_id`? `TypeId` can be both hashed and compared. 
Do we expect to have multiple types with same `get_name()`, that we should treat as a same protocol?
How this would handle rogue external type with `get_name() ==  "postgres"`?